### ISSUE TITLE
Multiple select statements do not work as expected

### ISSUE THREAD

**Issue by jackbond** [microsoftgraph/msgraph-sdk-dotnet#200]
We're attempting to migrate from Microsoft.Azure.ActiveDirectory.GraphClient to Microsoft.Graph but we're blocked by the fact that the Microsoft.Graph.User doesn't expose any sort of alternate email property.  What is the preferred approach for replacing the functionality of Microsoft.Azure.ActiveDirectory.GraphClient.IUser's OtherMails property? Thanks. 
[AB#7338](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7338)

**Comment by MIchaelMainer** [comment]
You are stuck using two client libraries and two endpoints until this functionality comes on to Microsoft Graph.

My approach would be to open a request on https://officespdev.uservoice.com/ and continue to use the AAD Graph client that targets the Azure AD endpoint until the functionality is exposed on the Microsoft Graph endpoint. Once that happens, the next release of the client library should have that functionality. Link to the user voice request from this issue so other people can find it.

**Comment by MIchaelMainer** [comment]
Unfortunately, you are stuck using two client libraries against two endpoints until this functionality is added to Microsoft Graph.

In the meantime, please open a request at https://officespdev.uservoice.com/ and link to your request from this issue.

**Comment by jackbond** [comment]
@MIchaelMainer Hi Michael. I appreciate your response (and optimism), but a quick search of the existing uservoice requests indicates that people have been requesting this feature for nearly a year.

https://officespdev.uservoice.com/forums/224641-feature-requests-and-feedback/suggestions/17449177-ability-to-update-the-user-s-email-aliases-proxya

https://officespdev.uservoice.com/forums/224641-feature-requests-and-feedback/suggestions/31987990-provide-access-to-the-alternate-email-othermails

Just to clarify for the Graph team (in case they are unaware), email is a popular means of communications on the internet, with nearly 200 BILLION emails sent every day.

Also, given the guidance issued [here](https://blogs.msdn.microsoft.com/aadgraphteam/2016/07/08/microsoft-graph-or-azure-ad-graph/), it's rather disappointing that this functionality hasn't already been delivered.

**Comment by MIchaelMainer** [comment]
I just realized after looking at the user voice request that you can GET the proxyAddresses with the **beta** endpoint. Does your scenario require just read or do you need write as well?

https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/user#properties

I agree, this functionality is core to many scenarios.

**Comment by jackbond** [comment]
LOL, ok, that's interesting, hiding under a different name? We definitely need both read and write. What is the beta endpoint? Also, ProxyAddresses throws an Exception if you try and set it at runtime, but technically it is a get/set property. I guess that allows a future implementation to enable this feature without being a breaking change.

**Comment by MIchaelMainer** [comment]
The default endpoint for Microsoft Graph client library is graph.microsoft.com/v1.0, the 'beta' endpoint is graph.microsoft.com/beta. The beta endpoint is superset of v1.0 and contains new functionality. It is advisable to not take dependencies on this endpoint as it does change. You can set the endpoint in the constructor for the GraphServiceClient object.  

Yes, the client library does implement proxyAddresses as a RW. We generate the client library from metadata and since the metadata doesn't specify RW, we go with the most permissive option.

**Comment by MIchaelMainer** [comment]
Well shoot, I was incomplete. It is available with the v1.0 endpoint and was generated for the client library.
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/src/Microsoft.Graph/Models/Generated/User.cs#L203. You should be able to use the library as is.

**Comment by jackbond** [comment]
If I create a user using Microsoft.Azure.ActiveDirectory.GraphClient.ActiveDirectoryClient and add to the OtherMails property, then retrieve the user with Microsoft.Graph, the emails don't show up in ProxyAddresses. Also if I instantiate a new Microsoft.Graph.User, ProxyAddresses is null, and if you attempt to set it, it raises an Exception stating it can't be set. Finally, since it is an IEnumerable, even if was being populated during retrieval, you wouldn't be able to add to it. Bottom line, the current implementation of the library isn't usable.

**Comment by MIchaelMainer** [comment]
You won't be able to set it. The docs say it read-only from MIcrosoft Graph. 

ProxyAddresses doesn't return as a default property in the v1.0 endpoint. You'll need to select it. 

`var user = await graphClient.Me.Request().Select("ProxyAddresses").GetAsync();`

Will this work on the read side for you?

**Comment by jackbond** [comment]
Are you only able to Select a single column at a time? Also, I Selected ProxyAddresses, and that doesn't appear to map to what the Azure library stored in OtherMails. I don't really care where the Microsoft.Graph library stores the data (I could write a utility in five minutes to migrate it), but it doesn't really feel like the OtherMails and ProxyAddresses are being saving to the same place in the underlying store. Finally, when I do Select ProxyAddresses, every other property ends up being null (which I somewhat understand.) Bottom line, this doesn't work for either reading or writing.

**Comment by MIchaelMainer** [comment]
You can select multiple columns, Use either a LINQ statement or chain multiple select statements.

**Comment by MIchaelMainer** [comment]
The multiple select statements is an issue. Another option is to add a collection of QueryOption objects in the Request(IEnumerable<Option>). 

1978815

**Comment by mlafleur** [comment]
Just stumbled upon this and wanted to clarify that you can select multiple columns by supply a comma-delimited list to `select`:

```cs
var user = await graphClient.Me
    .Request()
    .Select("displayName, id, mail, userPrincipalName,ProxyAddresses")
    .GetAsync();
```

**Comment by Tanjoodo** [comment]
Hey, I also just stumbled upon this and wanted to suggest that you can also use the type safe form using a LINQ statement for multiple columns by utilizing anonymous types as so:

``` CSharp
var user = await graphClient.Me
    .Request()
    .Select(u => new {u.DisplayName, u.Id, u.UserPrincipalName, u.ProxyAddresses})
    .GetAsync();
```

**Comment by StuartMeeks** [comment]
Why was this issue renamed? I have the same issue in that I need to access the OtherMails property. If ProxyAddresses and OtherMails don't used the same back end storage, then the initial issue still exists.

It also seems from the comments that the LINQ select chaining has a solution (i.e. chaining doesn't need to be supported since a comma separated list of fields will do the trick).

Can we get back to the original issue around the otherMails property?

**Comment by MIchaelMainer** [comment]
@StuartMeeks The OtherMails property support is a feature request for the service API. Service API feature requests should go to [UserVoice](https://officespdev.uservoice.com). We can't resolve the Service API feature requests or issues in this repo. Another option is via StackOverflow. I get the impression that StackOverflow is a better place to voice issues/feature requests for the service APIs. I don't find either of these options to be very satisfying . https://stackoverflow.com/search?q=%5Bmicrosoft-graph%5D+othermails - this feature appears to have demand.

What I should have done is create a new issue from the comment instead of repurposing this issue.

**Comment by maisarissi** [comment]
Hello everybody.

I'm closing this issue since the otherMails property was added to the Graph endpoint.
Check [update user documentation](https://docs.microsoft.com/en-us/graph/api/user-update?view=graph-rest-1.0&tabs=http#request-body) for more details.
